### DISCUSSION TITLE
How npm audit maintain its vulnerability database?

### DISCUSSION THREAD

**Discussion by nasifimtiazohi** [npm/feedback#172] [Registry] [answered]
Hi,

As part of our research, we are trying to understand how different advisory databases maintain and curate their vulnerability data. It'd be greatly helpful if we had some comments on the below questions: 

1. What are your sources for vulnerability data, e.g. NVD, OSS Index?
2. Do you have any process to discover open source vulnerabilities by yourselves, e.g. through monitoring bug repositories? If yes, is it possible to share with us a high level explanation of what you do?
3. When collecting vulnerability data from third-party databases (e.g. NVD), do you perform any curation and/or correction, e.g. discarding debated CVEs or correcting the affected version range? If yes, is it possible to share with us a high level explanation of what you do?

Sorry for the template questions. We're trying to reach out to different parties with the same set of questions so that we can study the answers systematically. Any insight would be greatly helpful. 

Nasif

**Comment by MylesBorins** [comment]
Hey @nasifimtiazohi I pinged some folks on the team to chime in.

**Comment by apple502j** [comment]
I *think* npm advisories are based on GH advisories (and previously, nodejs-ecosystem hackerone). They do not check NVD or Snyk afaik.

I contacted npm on Oct 22 (just after publication of CVE-2020-7750 and its Snyk advisory) and it took 2 weeks to get them listed as https://www.npmjs.com/advisories/1575 , and several vulnerabilities are still missing from the list. [Snyk](https://snyk.io/vuln?type=npm) is usually better when it comes to listing vulnerable packages, even though things like CVSS score tend to be inaccurate.

**Comment by mayakacz** [comment, marked as answer]
Hi again from https://github.community/t/how-github-maintains-its-advisory-database/148879 :wave:

1. The [npm Security Advisories](https://www.npmjs.com/advisories). This includes data reported by researchers directly to npm. Note that all of this data has [now been migrated to the GitHub Advisory Database](https://github.blog/changelog/2020-09-08-github-advisory-database-contains-all-npm-security-advisories/). These databases are now managed jointly, with the GitHub database as the source of truth. Updates to the GitHub database are pushed to the npm database. npm audit still uses the npm database.
2. No, prior to the npm acquisition, this was primarily based on third-party reports to npm. Now this includes issues found by GitHub, including GitHub Security Lab.
3. Since the npm and GitHub databases are managed jointly, the same curation applies. See a description in this blog post: https://github.blog/2019-12-11-behind-the-scenes-github-vulnerability-alerts/
